### PR TITLE
Regression preprocessing function

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Exporting regression preprocessing function for usage outside of Analytics View.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/build_lm.R [hunk 1]
```diff
@@ -255,6 +255,148 @@ map_terms_to_orig <- function(terms, mapping) {
...
+  # cols will be filtered to remove invalid columns
+  cols <- predictor_cols
+  for (col in predictor_cols) {
+    if(all(is.na(df[[col]]) | is.infinite(df[[col]]))){
+      # remove columns if they are all NA or Inf
+      cols <- setdiff(cols, col)
+      df[[col]] <- NULL # drop the column so that SMOTE will not see it. 
+    }
+  }
+
```
Comment (kei51e) [inline R/build_lm.R:271]: Stop here if no predictors are left.

File: R/build_lm.R [hunk 2]
```diff
@@ -255,6 +255,148 @@ map_terms_to_orig <- function(terms, mapping) {
...
+        df[[hour_col]] <- factor(lubridate::hour(df[[col]])) # treat hour as category
+      }
+      df[[col]] <- NULL # drop original Date/POSIXct column to pass SMOTE later.
+    } else if(is.factor(df[[col]])) {
+      df[[col]] <- forcats::fct_drop(df[[col]]) # margins::marginal_effects() fails if unused factor level exists. Drop them to avoid it.
+      # 1. if the data is factor, respect the levels and keep first 10 levels, and make others "Others" level.
+      # 2. if the data is ordered factor, turn it into unordered. For ordered factor,
+      #    lm/glm takes polynomial terms (Linear, Quadratic, Cubic, and so on) and use them as variables,
+      #    which we do not want for this function.
+      if (length(levels(df[[col]])) >= predictor_n + 2) {
```
Comment (kei51e) [inline R/build_lm.R:360]: Why +2?

File: R/build_lm.R (hunk 1 above)
Comment (hideaki) [reply R/build_lm.R:274]: Added error handling.

File: R/build_lm.R (hunk 2 above)
Comment (hideaki) [reply R/build_lm.R]: Added comment.

**Pull request merged**